src/components/RippleButton/RippleButton.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useRef, useState } from 'react';
+import {
+  View,
+  TouchableOpacity,
+  Animated,
+  GestureResponderEvent,
+  LayoutChangeEvent,
+} from 'react-native';
+import type { RippleProps } from './types';
+import styles from './RippleStyles';
+
+const RippleButton: React.FC<RippleProps> = ({
+  style,
+  onPress,
+  animatedDuration = 500,
+  rippleColor = 'rgba(0,0,0,0.2)',
+  children,
+}) => {
+  const translation: Animated.Value = useRef(new Animated.Value(0)).current;
```

Comment:
@nilesh-simform We should create a custom hook inside RippleButton to move this animation logic there.
e.g RipplerButton/useRippleButton.ts